docs: added release apache apisix 2.7 English version blog (#397)
iresty/iresty.github.io: website/blog/2021-06-29-release-apache-apisix-2.7-en.md

---
title: "Release Apache APISIX 2.7.0"
author: Serendipity96
authorURL: "https://github.com/Serendipity96"
authorImageURL: "https://avatars.githubusercontent.com/u/23514812?v=4"
---
> [@Serendipity96](https://github.com/Serendipity96), Apache APISIX contributor from [Shenzhen Zhiliu Technology Co.](https://www.apiseven.com/)
>
<!--truncate-->

Apache APISIX 2.7.0 has been released! Welcome to download and use it.

> This version supports multi-language plugins, enhances the four-layer TCP proxy, and enhances Nginx configuration. More than 20 developers participated, and more than 70 pull requests have been submitted. The following is an introduction to the key features.

Reviewer: ```suggestion
This version supports multi-language plugins, enhances the four-layer TCP proxy and Nginx configuration. More than 20 developers participated, and 70 plus pull requests have been submitted. The following is an introduction to the key features.
```

Author: "plus" is better than "more than"?

Reviewer: it's a bit duplicated: `"More than" 20 developers participated, and "more than" 70 pull requests have been submitted`

I think changing it to another expression could be fluent

## Release Notes

### Multi-language plugin

> With the release of the first version of apisix-java-plugin-runner,  and apisix-go-plugin-runner completes main function,  the multi-language plugin of Apache APISIX supports the two most widely used back-end programming languages. If you are still worried that the development of Apache APISIX plugin will be limited to Lua ecosystem, please try our multi-language plugin runner to develop Java / Go plugins.

Reviewer 2: ```suggestion
With the release of the first version of apisix-java-plugin-runner, and apisix-go-plugin-runner completes its main functions, the multi-language plugin of Apache APISIX supports the two most widely used back-end programming languages. If you are still worried that the development of Apache APISIX plugin will be limited to Lua ecosystem, please try our multi-language plugin runner to develop Java / Go plugins.
```

Reviewer 3: ```suggestion
With the release of the first version of apisix-java-plugin-runner,  and apisix-go-plugin-runner completes main function, the multi-language plugin of Apache APISIX supports the two most widely used back-end programming languages. If you are still think that the development of Apache APISIX plugin will be limited to Lua ecosystem, please try our multi-language plugin runner to develop Java / Go plugins.
```

Reviewer 3: worried seems to be an out of context word here IMO

Reviewer 2: `worried` is fine... The original Chinese text uses the same word.

Reviewer: ```suggestion
With the first release of apisix-java-plugin-runner, and the completion of apisix-go-plugin-runner's main functions, the multi-language plugin of Apache APISIX now supports the two most widely used back-end programming languages. If you are still worried that the plugin development of Apache APISIX will be limited to Lua ecosystem, welcome to use our multi-language plugin runner to develop your own Java / Go plugins!
```

Reviewer 4: Agree with @iamayushdas here. I guess "hesitant" should be the right word choice here, regardless of language @spacewander as in "If you are still **hesitant** that the plugin development of Apache APISIX will be limited to Lua ecosystem". What do you think?

Reviewer 2: Using "hesitant" is a good choice. But I think being strict with the original text is better.

Reviewer 4: Oh, Makes sense 👍

> With the release of the first version of apisix-java-plugin-runner,  and apisix-go-plugin-runner completes main function,  the multi-language plugin of Apache APISIX supports the two most widely used back-end programming languages. If you are still worried that the development of Apache APISIX plugin will be limited to Lua ecosystem, please try our multi-language plugin runner to develop Java / Go plugins.

Reviewer 4: ```suggestion
With the release of the first version of apisix-java-plugin-runner,  and apisix-go-plugin-runner completes main function,  the multi-language plugin of Apache APISIX supports the two most widely used back-end programming languages. If you are still hesitant that the development of Apache APISIX plugin will be limited to Lua ecosystem, please try our multi-language plugin runner to develop Java / Go plugins.
```

Reviewer 3: Exactly , this is what i was saying

Reviewer 2: I don't want to waste time arguing a word. This is no English course. As for translation, we should not have so much secondary creation.

### Enhanced four-layer TCP proxy

In version 2.7, we have developed new features of TCP proxy, including:

- Allow domain name configuration in upstream
- Allow mqtt-proxy plugin to configure domain name

> - Support receiving TLS over TCP connection, the certificate can be dynamically configured like HTTPS certificate

Reviewer: ```suggestion
- Support for receiving TLS over TCP connections, the certificate of which can be dynamically configured just like HTTPS certificate
```

> - Route rules  based on SNI

Reviewer: ```suggestion
- SNI-based route rules
```

> - Verify client certificate dynamically

Reviewer: ```suggestion
- Dynamic verification of client certificates
```

> In subsequent versions, we will continue to improve TCP proxy,  stay tuned!

Reviewer: ```suggestion
In future releases, we will continue to improve TCP proxy, so stay tuned!
```

> In subsequent versions, we will continue to  improve TCP proxy,  stay tuned!

Reviewer 5: ```suggestion
In subsequent versions, we will continue to improve TCP proxy,  stay tuned!
```

### Enhanced Nginx configuration

> We hope to set more and more Nginx configurations dynamically, so we add our own patches and Nginx C modules to enhance native Nginx.

Reviewer: ```suggestion
We hope to dynamically set more and more Nginx configurations, so we add our own patches and Nginx C modules to enhance the native Nginx.
```

The following new features are currently included:

> - Set mTLS dynamically
> - Set client_max_body_size dynamically

Reviewer: ```suggestion
- Dynamically set mTLS 
- Dynamically set client_max_body_size
```

> In subsequent versions, we will continue to set the following Nginx configurations to dynamically:

Reviewer: ```suggestion
In future releases, we will continue to allow the following Nginx configurations to be set dynamically:
```

- upstream keepalive
- gzip
- real_ip
- proxy_max_temp_file_size

## Download

Download Apache APISIX 2.7.0-Release

- Source code: please visit [download page](https://apisix.apache.org/downloads/)

> - Binary installation package: please visit [installation guide](https://apisix.apache.org/zh/docs/apisix/how-to-build/)

Reviewer 5: ```suggestion
- Binary installation package: please visit [Installation Guide](https://apisix.apache.org/zh/docs/apisix/how-to-build/)
```
